File: 백준/Silver/11723. 집합/집합.py
import sys
import copy

n = int(input())

s = set()
for _ in range(n):
    command = sys.stdin.readline().rstrip("\n").split()
    
    if command[0] == 'add':
        s.add(command[1])
    elif command[0] == "remove":
        if command[1] in s:
            s.discard(command[1])
    elif command[0] == "check":
        print(1 if command[1] in s else 0)
    elif command[0] == "toggle":
        if command[1] in s:
            s.discard(command[1])
        else:
            s.add(command[1])
    elif command[0] == "all":
        s = {'1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20'}
    elif command[0] == "empty":
        s = set()

File: 백준/Silver/11723. 집합/집합.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int set = 0;
        int m = Integer.parseInt(br.readLine());
        StringBuilder sb = new StringBuilder();

        for(int i=0; i<m; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());

            String command = st.nextToken();

            if (command.equals("add")) {
                int num = Integer.parseInt(st.nextToken());

                set |= (1 << num);
            }

            else if (command.equals("remove")) {
                int num = Integer.parseInt(st.nextToken());

                set &= ~(1 << num);
            }

            else if (command.equals("check")) {
                int num = Integer.parseInt(st.nextToken());

                if ((set & (1 << num)) != 0) {
                    sb.append("1\n");
                }
                else {
                    sb.append("0\n");
                }
            }

            else if (command.equals("toggle")) {
                int num = Integer.parseInt(st.nextToken());

                set ^= (1 << num);
            }

            else if (command.equals("all")) {
                set = (1 << 21) - 1;
            }

            else if (command.equals("empty")) {
                set = 0;
            }
        }
        System.out.println(sb);
    }
    
}
